t/apicast-policy-upstream-connection.t

```diff
@@ -1,6 +1,8 @@
 use lib 't';
 use Test::APIcast::Blackbox 'no_plan';
 
+require("http_proxy.pl");
```

Comment:
one comment not related to this PR: I am thinking about running one http proxy (tinyproxy for example) in the docker compose env we run for the e2e tests and get rid of this custom, perl implemented, proxy we are using. Currently, in addition to the apicast dev container, there is a redis one. 

Same can be done for the camel proxy. 

For backend, upstream, it is very handy to do some asserts so, "outsourcing" upstream may not be convenient. 

wdyt?

Reply:
I think there are a few tests that watch the output of this proxy module, ie headers.  So we will need to rewrite a few tests, but on the flip side we get more reliable test (may be), ie avoid the case where e2e tests run before the proxy module is ready 

We will need at least 4 containers for both `dev` and `ci` image. 
  *  Normal proxy
  * A camel proxy
  * A camel proxy support HTTPS
  * A proxy that accept BasicAuth

I had a similar problem with #1414 where I need 4 more containers just to be able to test redis sentinel.

As long as it works well and is easy to maintain, I really don't mind adding additional pods.

Reply:
Looks like we have a TODO list. We can implement step by step and remove the perl based proxy when it is fully being replaced by other proxies. 

PS: `dev` and`ci` image are currently the same one